akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1064,6 +1064,29 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
     new Flow(delegate.keepAlive(maxIdle, () ⇒ injectedElem.create()))
 
   /**
+   * Sends elements downstream after each time interval. In other words, this stage set the maximum rate
+   * for emitting messages.
+   *
+   * This stage is `zipWith` combination of upstream connection and [[akka.stream.scaladsl.Source#tick]]
+   * that triggers sending downstream messages.
+   *
+   * General recommendation is to use upstream `buffer` with predefined [[akka.stream.OverflowStrategy]]
+   * that defines behaviour on buffer overflow.
+   *
+   * '''Emits when''' upstream emits an element and configured time elapsed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @see [[#buffer]], [[akka.stream.scaladsl.Source#tick]].
+   */
+  def throttle(initialDelay: FiniteDuration, interval: FiniteDuration): javadsl.Flow[In, Out, Mat] =
```

Comment:
+1 for token bucket.in fact one of my frinds in suning is using this with akka to implement throttling.